setup.py

```diff
@@ -47,194 +47,390 @@
+        build_py.run(self)
+
+        # dst_root is the root of the `executorch` module in the output package
+        # directory. build_lib is the platform-independent root of the output
+        # package, and will look like `pip-out/lib`. It can contain multiple
+        # python packages, so be sure to copy the files into the `executorch`
+        # package subdirectory.
+        dst_root = os.path.join(self.build_lib, self.get_package_dir("executorch"))
+
+        # Manually copy files into the output package directory. These are
+        # typically python "resource" files that will live alongside the python
+        # code that uses them.
+        src_to_dst = [
+            # TODO(dbort): See if we can add a custom pyproject.toml section for
+            # these, instead of hard-coding them here. See
+            # https://setuptools.pypa.io/en/latest/userguide/extension.html
+            ("schema/scalar_type.fbs", "exir/_serialize/scalar_type.fbs"),
+            ("schema/program.fbs", "exir/_serialize/program.fbs"),
+            (
+                "sdk/bundled_program/schema/bundled_program_schema.fbs",
+                "sdk/bundled_program/serialize/bundled_program_schema.fbs",
+            ),
+            (
+                "sdk/bundled_program/schema/scalar_type.fbs",
+                "sdk/bundled_program/serialize/scalar_type.fbs",
+            ),
+        ]
+        for src, dst in src_to_dst:
+            dst = os.path.join(dst_root, dst)
+
+            # When modifying the filesystem, use the self.* methods defined by
+            # Command to benefit from the same logging and dry_run logic as
+            # setuptools.
+
+            # Ensure that the destination directory exists.
+            self.mkpath(os.path.dirname(dst))
+            # Follow the example of the base build_py class by not preserving
+            # the mode. This ensures that the output file is read/write even if
+            # the input file is read-only.
+            self.copy_file(src, dst, preserve_mode=False)
+
+
+# TODO(dbort): For editable wheels, may need to update get_source_files(),
+# get_outputs(), and get_output_mapping() to satisfy
+# https://setuptools.pypa.io/en/latest/userguide/extension.html#setuptools.command.build.SubCommand.get_output_mapping
+
+
+class CustomBuild(build):
+    def initialize_options(self):
+        super().initialize_options()
+        # The default build_base directory is called "build", but we have a
+        # top-level directory with that name. Setting build_base in setup()
+        # doesn't affect this, so override the core build command.
+        #
+        # See build.initialize_options() in
+        # setuptools/_distutils/command/build.py for the default.
+        self.build_base = "pip-out"
+
+        # Default build parallelism based on number of cores, but allow
+        # overriding through the environment.
+        default_parallel = str(os.cpu_count() - 1)
```

Comment:
Nice change. This should speed up builds significantly on high core count systems.